File: e1b/examples/count.py
# Counts to 10 million and times it
# Printing out every 1 million
# Type "python count.py" to run the program

from time import time

# Record the start time
start = time()
# Set our max number for counting
n = 10_000_000
# Loop through each number between 1 and n
for i in range(n):
    # "%" is "modulo" -- the remainder after division
    if i % (n / 10) == 0:
        # Show us where we are in the loop
        print(i)
# Record the end time
end = time()
# See how long it took
print(f"Time in Python: {end - start:.4} seconds")

File: e1b/examples/count.c
// Counts to 10 million and times it
// printing out every 1 million
// type "./count" to run
// type "make count" if you change the number

#include <stdio.h>
#include <time.h>

int main(void)
{
    // Record the start time
    clock_t start = clock();
    // Set our max number for counting
    int n = 10000000;
    // Loop through each number between 1 and n
    for (int i = 0; i < n; i++)
    {
        // "%" is "modulo" -- the remainder after division
        if (i % (n / 10) == 0)
        {
            // Show us where we are in the loop
            printf("%i\n", i);
        }
    }
    // Record the end time
    clock_t end = clock();
    // See how long it took
    printf("Time in C: %f seconds \n", (double)(end - start) / 1000000);
}
